docs/notes/02-packaging.md

```diff
@@ -142,6 +117,89 @@ testing notes](05-linters-and-tests.md) for more info.
+
+## Github Actions
+
+We deploy to PyPI every release, and we follow [semantic versioning](https://semver.org/) for release labeling (see [workflow notes](00-workflow.md) for more details on when to release and how this labeling works; this repo's `CONTRIBUTING.md` document also includes some suggested language for explaining the procedure to ccontributors). We do this by making use of continuous integration (CI) on github actions. See the [CI](06-ci.md) notes for more on CI and Github actions in general; in the following sections we discuss specifically how to use our `deploy` actions.
+
+We provide two different Github actions for handling building and deployment: `deploy-pure-python.yml` and `deploy-cibw.yml` (`cibw` = "CI build wheel"). If your package is pure python, as described earlier in this note, use `deploy-pure-python.yml`, otherwise use `deploy-cibw.yml`. If you're not sure, manually build your package as described [earlier](#build) (`python -m build --wheel`) and then `ls dist/*.whl` to examine your built wheels. If your output looks like `package-version-py3-none-any.whl`, it's pure python. If it includes the name of your OS in the filename, it is package-specific and you should use `deploy-cibw.yml`.
+
+### Setup 
+
+Do the following steps, regardless of whether you're using the pure python or CI build wheel action.
+
+1. Create an account on both [PyPI](https://pypi.org/) and [Test PyPI](https://test.pypi.org/). 
+2. Follow the steps on the [pypi docs](https://docs.pypi.org/trusted-publishers/creating-a-project-through-oidc/) to create a PyPI project with a trusted publisher. **NOTE:** do this for both PyPI and Test PyPI --- it's recommended you always deploy to Test PyPI at first, to make sure things look the way you want, and then switch to PyPI when you're ready.
+3. Copy the relevant `deploy` file to your repository.
+
+### Understanding the action file
+
+#### Pure python
+
+The action defined in `.github/workflows/deploy-pure-python.yml` builds and deploys a pure python project. This section is based on [this guide](https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/) from the python packaging authority and makes use of [their action](https://github.com/pypa/gh-action-pypi-publish).
+
+This action gets triggered on every github release and does the following:
```

Comment:
PyPI and Test PyPI are completely separate indices, so there's no problem with conflicting names or tags. Thus, to deploy with the same tag, I'd remove the`with` lines, delete the tag and go again.

Reply:
I would be worried about putting too much string-parsing logic like that in the job. My intention was that people would go to Test PyPI just in the beginning, to make sure the action is working and it looks like they want, and then deploy right to PyPI from then on, rather than always deploying to both.

The only use case I can think of for why they'd want to go back to TestPyPI afterwards would be if they're adding/removing a bunch of files and want to check it all again. But in that case, they could modify the test line at the end of build job or do it manually (e.g., build the wheel locally and investigate it / upload to Test PyPI)